sbol2/componentdefinition.py

```diff
@@ -134,7 +129,46 @@ def __init__(self, uri=URIRef("example"),
         self.sequenceConstraints = OwnedObject(self, SBOL_SEQUENCE_CONSTRAINTS,
                                                SequenceConstraint,
                                                '0', '*', None)
-        self._hidden_properties = [SBOL_SEQUENCE]
+        # This caches a sequence object so that it can be added
+        # to a document later
+        self._sequence_cache: Union[Sequence, None] = None
+
+    @property
+    def sequence(self):
+        seqs = self.sequences
+        if not seqs:
+            return None
+        seq_uri = seqs[0]
+        if self._sequence_cache and self._sequence_cache.identity == seq_uri:
+            return self._sequence_cache
+        # Not in the cache, try to look up in the document
+        if self.doc:
+            try:
+                return self.doc.find(seq_uri)
+            except NotImplementedError:
+                raise
+
+    @sequence.setter
+    def sequence(self, sequence: Union[Sequence, None]):
+        if not sequence:
+            # TODO: How to remove an orphaned sequence from the document?
```

Comment:
I don't think this is an issue.  Even if the Sequence is set to None, it's still OK for that Sequence to persist in the Document (e.g., maybe the user wants to re-use the Sequence with some other ComponentDefinition)